test/connectors/FinancialDataConnectorSpec.scala

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import base.SpecBase
+import com.github.tomakehurst.wiremock.client.WireMock._
+import models.Period
+import models.financialdata.{FinancialData, FinancialTransaction, Item}
+import models.payments.{Payment, PaymentStatus, PrepareData}
+import play.api.Application
+import play.api.test.Helpers.running
+import uk.gov.hmrc.http.HeaderCarrier
+
+import java.time.{LocalDate, Month, ZoneOffset, ZonedDateTime}
+
+class FinancialDataConnectorSpec extends SpecBase with WireMockHelper with FinancialDataConnectorFixture {
+  implicit private lazy val hc: HeaderCarrier = HeaderCarrier()
+
+  private def application: Application = {
+    applicationBuilder()
+      .configure("microservice.services.ioss-returns.port" -> server.port)
+      .build()
+  }
+
+  private val now = zonedNow.toLocalDate
+  "getFinancialData" - {
+    def financialDataUrl(commencementDate: LocalDate) = s"/ioss-returns/financial-data/get/$commencementDate"
+
+    "when the server returns OK and a recognised payload" - {
+      "must return a FinancialDataResponse" in {
+        val app = application
+
+        server.stubFor(
+          get(urlEqualTo(financialDataUrl(now)))
+            .willReturn(ok(responseJsonFinancialData))
+        )
+
+        running(app) {
+          val connector = app.injector.instanceOf[FinancialDataConnector]
+          val result = connector.getFinancialData(now).futureValue
+
+          result mustEqual expectedResultFinancialData
+        }
+      }
+    }
+  }
+
+  "prepareData" - {
+    def prepareDataUrl() = s"/ioss-returns/financial-data/prepare"
+
+    "when the server returns OK and a recognised payload" - {
+      "must return a FinancialDataResponse" in {
+        val app = application
+
+        server.stubFor(
+          get(urlEqualTo(prepareDataUrl()))
```

Comment:
Inline? `get(urlEqualTo("/ioss-returns/financial-data/prepare"))`

Reply:
You're right, as it's used only once. Done.